digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/JunctionPointDAO.scala

```diff
@@ -20,6 +20,26 @@ class JunctionPointDAO extends BaseDAO {
 
   val dateFormatter: DateTimeFormatter = ISODateTimeFormat.basicDate()
 
+  val junctionPointQuery =
+    """
+      SELECT JP.ID, JP.BEFORE_AFTER, JP.ROADWAY_POINT_ID, JP.JUNCTION_ID, J.START_DATE, J.END_DATE, JP.VALID_FROM, JP.VALID_TO, JP.CREATED_BY, JP.CREATED_TIME,
+      RP.ROADWAY_NUMBER, RP.ADDR_M, RW.ROAD_NUMBER, RW.ROAD_PART_NUMBER, RW.TRACK, RW.DISCONTINUITY
+      FROM JUNCTION_POINT JP
+      JOIN JUNCTION J ON (J.ID = JP.JUNCTION_ID AND J.VALID_TO IS NULL AND J.END_DATE IS NULL)
+      JOIN ROADWAY_POINT RP ON (RP.ID = ROADWAY_POINT_ID)
+      JOIN ROADWAY RW ON (RW.ROADWAY_NUMBER = RP.ROADWAY_NUMBER)
+    """
+
+  val junctionPointHistoryQuery =
+    """
+      SELECT JP.ID, JP.BEFORE_AFTER, JP.ROADWAY_POINT_ID, JP.JUNCTION_ID, J.START_DATE, J.END_DATE, JP.VALID_FROM, JP.VALID_TO, JP.CREATED_BY, JP.CREATED_TIME,
+      RP.ROADWAY_NUMBER, RP.ADDR_M, RW.ROAD_NUMBER, RW.ROAD_PART_NUMBER, RW.TRACK, RW.DISCONTINUITY
+      FROM JUNCTION_POINT JP
+      JOIN JUNCTION J ON (J.ID = JP.JUNCTION_ID)
+      JOIN ROADWAY_POINT RP ON (RP.ID = ROADWAY_POINT_ID)
+      JOIN ROADWAY RW ON (RW.ROADWAY_NUMBER = RP.ROADWAY_NUMBER)
```

Comment:
`JOIN ROADWAY RW ON (RW.ROADWAY_NUMBER = RP.ROADWAY_NUMBER AND RW.VALID_TO IS NULL AND RW.END_DATE IS NULL)`

Reply:
This query is currently being used only for tests, and we didn't add the clause to the roadway so we make sure the points for the terminated roadway are also terminated.